auto_nag/db.py

```diff
@@ -0,0 +1,422 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import calendar
+import csv
+import dateutil.parser
+from filelock import FileLock
+import json
+from libmozdata import utils as lmdutils
+import os
+import six
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import create_engine
+from sqlalchemy.orm import relationship, sessionmaker
+from sqlalchemy import Column, ForeignKey, Integer, String
+
+from auto_nag import logger, utils
+from auto_nag.history import History
+
+
+Base = declarative_base()
+lock_path = utils.get_config('common', 'lock')
+db_path = utils.get_config('common', 'database')
+engine = create_engine(db_path)
+DBSession = sessionmaker(bind=engine)
+Base.metadata.bind = engine
+session = DBSession()
+
+
+def clear():
+    Base.metadata.drop_all()
+    session.commit()
+
+
+def create():
+    if not engine.dialect.has_table(engine, 'tools'):
+        Base.metadata.create_all(engine)
+        init()
+
+
+def init():
+    hist = History().get()
+    logger.info('Put history in db: start...')
+    BugChange.read_dict(hist)
+    logger.info('Put history in db: end.')
+
+
+def get_ts(date, default=0):
+    if isinstance(date, six.integer_types):
+        return date
+    if date:
+        if isinstance(date, six.string_types):
+            date = dateutil.parser.parse(date)
+        date = int(calendar.timegm(date.timetuple()))
+        return date
+    if default == 'now':
+        return lmdutils.get_timestamp('now')
+    return default
+
+
+class Tool(Base):
+    __tablename__ = 'tools'
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    name = Column(String(128), unique=True)
+    bugchanges = relationship('BugChange', backref='tool')
+    sentmails = relationship('SentEmail', backref='tool')
+
+    def __init__(self, name):
+        self.name = name
+
+    @staticmethod
+    def get(name):
+        e = session.query(Tool).filter(Tool.name == name).first()
```

Comment:
I would rather use [one()](https://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.one) and create in the exception handler